python/lsst/obs/base/cli/cmd/write_curated_calibrations.py

```diff
@@ -0,0 +1,60 @@
+# This file is part of obs_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import logging
+
+from lsst.daf.butler.cli.opt import repo_option, run_option
+from lsst.obs.base.cli.opt import instrument_option
+from lsst.daf.butler import Butler
+from lsst.obs.base import Instrument
+from lsst.utils import doImport
+
+__all__ = ['write_curated_calibrations']
+
+
+@click.command(name="write-curated-calibrations")
+@instrument_option(required=True)
+@repo_option(required=True)
+@run_option(required=True)
+@click.pass_context
+def write_curated_calibrations(context, repo, instrument, output_run):
+    """Add an instrument's curated calibrations to the data repository.
+    """
+    butler = Butler(repo, writeable=True, run=output_run)
+    if "." in instrument:
+        try:
+            instr = doImport(instrument)
+        except Exception as err:
+            logging.critical("Could not import instrument: %s. Failed with exception: %s", instrument, err)
+            context.exit(1)
+    else:
+        try:
+            instr = Instrument.fromName(instrument, butler.registry)
+        except Exception as err:
+            logging.critical("Could not get instrument: %s. Failed with exception: %s", instrument, err)
+            context.exit(1)
+    try:
+        instr = instr()
+    except Exception as err:
+        logging.critical("Could not instantiate instrument %s. Failed with exception: %s", instr, err)
+        context.exit(1)
```

Comment:
This code is going to be extremely common and will be needed in multiple commands. It should be put somewhere that can be picked up the commands easily (also fromName returns an instance and not a class).